livy/src/main/java/org/apache/zeppelin/livy/LivyHelper.java

```diff
@@ -293,16 +295,20 @@ private InterpreterResult getResultFromMap(Map jsonMap) {
   }
 
   private Map executeCommand(String lines, InterpreterContext context,
-                             Map<String, Integer> userSessionMap) throws Exception {
-    String json = executeHTTP(property.get("zeppelin.livy.url") + "/sessions/"
+                             Map<String, Integer> userSessionMap) throws LivyException {
+    String json;
+    try {
+      json = executeHTTP(property.get("zeppelin.livy.url") + "/sessions/"
             + userSessionMap.get(context.getAuthenticationInfo().getUser())
             + "/statements",
         "POST",
         "{\"code\": \"" + StringEscapeUtils.escapeJson(lines) + "\"}",
         context.getParagraphId());
-    if (json.matches("^(\")?Session (\'[0-9]\' )?not found(.?\"?)$")) {
-      throw new Exception("Exception: Session not found, Livy server would have restarted, " +
-          "or lost session.");
+    } catch (Exception e) {
+      throw new LivyException("Error executing command in Livy", e);
+    }
+    if (json == null || json.matches("^(\")?Session (\'[0-9]\' )?not found(.?\"?)$")) {
```

Comment:
need to be change like this, isn't it?
json.matches("^(\")?Session (\'[0-9]\' )?not found(.?\"?)$")